concreteproperties/concrete_section.py

```diff
@@ -1036,186 +1036,215 @@ def calculate_ultimate_section_actions(
-                return d_ext * (eps_cu) / (cp[1] * eps_sy + eps_cu)
-            # provided axial force
-            elif cp[0] == "N":
-                ult_res = self.ultimate_bending_capacity(theta=theta, n=cp[1])
-                return ult_res.d_n
-            # zero curvature
-            elif cp[0] == "kappa0":
-                return 2 * d_t  # sufficient depth to capture rectangular block
-            # control point type not valid
-            else:
-                raise ValueError(
-                    "First value of control_point tuple must be D, d_n, fy, N or kappa0."
-                )
+        # generate list of neutral axis depths/axial forces to analyse
+        # if we are spacing by axial force
+        if n_spacing:
+            # get axial force of the limits
+            start_res = self.calculate_ultimate_section_actions(
+                d_n=limits_dn[0],
+                ultimate_results=res.UltimateBendingResults(theta=theta),
+            )
+            end_res = self.calculate_ultimate_section_actions(
+                d_n=limits_dn[1],
+                ultimate_results=res.UltimateBendingResults(theta=theta),
+            )
 
-        # see if a kappa0 was used
-        has_kappa0 = False
+            # correct n_spacing if sign is wrong
+            if start_res.n > end_res.n and n_spacing > 0:
+                n_spacing = -1.0 * n_spacing
+            elif start_res.n < end_res.n and n_spacing < 0:
+                n_spacing = -1.0 * n_spacing
 
-        for cp in control_points:
-            if cp[0] == "kappa0":
-                has_kappa0 = True
-                break
-
-        # generate list of neutral axis depths to analyse and list of labels to save
-        d_n_list = []
-        label_list = []
-        start_d_n = 0
-        end_d_n = 0
-
-        for idx, n_pt in enumerate(n_points):
-            # get netural axis depths from control_points
-            start_d_n = decode_d_n(control_points[idx])
-            end_d_n = decode_d_n(control_points[idx + 1])
-
-            # generate list of neutral axis depths for this interval
-            d_n_list.extend(
-                np.linspace(
-                    start=start_d_n, stop=end_d_n, num=n_pt - 1, endpoint=False
-                ).tolist()
-            )
+            # generate list of axial forces
+            analysis_list = np.arange(
+                start=start_res.n, stop=end_res.n, step=n_spacing, dtype=float
+            ).tolist()
 
-            # add labels
-            label_list.append(labels[idx])
-            label_list.extend([None] * (n_pt - 2))
+            # add end axial force
+            analysis_list.append(end_res.n)
+        else:
+            # check for infinity in limits - this will not work with linspace
+            # for sake of distributing neutral axes let kappa0 ~= 2 * D
+            if limits_dn[0] == inf:
+                start = 2 * d_t
+            else:
+                start = limits_dn[0]
```

Comment:
should there be a check if the `limits` are supplied the wrong way around prior to calculating `limits_dn` above and swap or just use `max(limits)` and `min(limits)` so the user entered order is not important? (rather than implying `limits[0]` and `limits[1]` are max and min)

Reply:
There is another ``if`` statement below this that handles this. I think it currently works letting you enter limits in either order. Let me know if you have an example that breaks this.